test/studies/shootout/meteor/kbrady/meteor-parallel-alt.chpl

```diff
@@ -1,23 +1,29 @@
 /* The Computer Language Benchmarks Game
- * http://benchmarksgame.alioth.debian.org/
- *
- * contributed by Kyle Brady and Ben Albrecht
- * derived from the C++ implementation by Stefan Westen which states it was
- * loosely based on Ben St. John's and Kevin Barnes' implementation
+   http://benchmarksgame.alioth.debian.org/
+
+   contributed by Kyle Brady and Ben Albrecht
+   derived from the C++ implementation by Stefan Westen which states it was
+   loosely based on Ben St. John's and Kevin Barnes' implementation
  */
 
 use BitOps;
 
-// Common ranges encapsulated as domains for simplicity
-const boardDom = {0..49},
-      permutationsDom = {0..8191},
-      piecesDom = {0..9};
+param boardCells = 50,          // Number of cells on board
+      boardWidth = 5,           // Number of cells along x-axis
+      boardHeight = 10,         // Number of cells along y-axis
+      permutations = 8192,      // Number of permutations stored as masks
+      piecePermutations = 12,   // Number of permutations for a single piece
+      numPieces = 10,           // Number of puzzle pieces
+      pieceCells = 5;           // Number of cells that make up a puzzle piece
+
+const boardDom = {0..#boardCells},
+      piecesDom = {0..#numPieces};
 
 // Arrays of masks to store all pieces and their possible configurations
-var allMasks: [permutationsDom] int,
-    maskStart: [boardDom][0..7] int;
+var allMasks: [0..#permutations] int,
+    maskStart: [boardDom][0..#numPieces-2] int;
 
-// Arrays of min, max, and number of solutions
+// Arrays of min, max, and integer storing number of solutions
```

Comment:
I would rewrite this as "Arrays of min and max, and an integer storing the number of solutions".  Otherwise, it parses weirdly for me